src/main/java/com/temat29zad1/user/PasswordResetService.java

```diff
@@ -0,0 +1,54 @@
+package com.temat29zad1.user;
+
+import org.springframework.stereotype.Service;
+
+import java.time.LocalDateTime;
+import java.util.Optional;
+import java.util.UUID;
+
+@Service
+public class PasswordResetService {
+
+    private final PasswordResetTokenRepository passwordResetTokenRepository;
+    private final UserRepository userRepository;
+
+    public PasswordResetService(PasswordResetTokenRepository passwordResetTokenRepository, UserRepository userRepository) {
+        this.passwordResetTokenRepository = passwordResetTokenRepository;
+        this.userRepository = userRepository;
+    }
+
+    public String returnTokenIfValidated(String token) {
+        Optional<PasswordResetToken> passToken = passwordResetTokenRepository.findPasswordResetTokenByToken(token);
+        if (passToken.isEmpty() || passToken.get().getExpiryDateTime().isBefore(LocalDateTime.now())) {
+            return "";
+        }
+        return passToken.get().getToken();
+    }
+
+    public void deleteExpiredTokenIfExists(Long id) {
+        Optional<PasswordResetToken> byUserId = passwordResetTokenRepository.findByUser_Id(id);
+        if (byUserId.isPresent()) {
+            if (returnTokenIfValidated(byUserId.get().getToken()).isEmpty()) {
+                passwordResetTokenRepository.deleteTokenByToken(byUserId.get().getToken());
+            }
+        }
+    }
+
+    public String generatePasswordResetToken(String email) {
```

Comment:
Jeżeli metoda nazywa się "generate...Token", to powinna utworzyć i zwrócić token, a nie url.
Tworzeniem URLa zajmij się raczej na poziomie metody do wysyłania maila.
Zerknij na metodę `ServletUriComponentsBuilder.fromCurrentContextPath()` - zwróci Ci ona context path, czyli jeżeli np. adres zmieni się na localhost:8082, czy coś innego, to będzie to bardziej uniwersalne. Docelowo trzeba by też uwzględnić np. domenę